staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types_jsonschema.go

```diff
@@ -62,6 +62,19 @@ type JSON struct {
 	Raw []byte `protobuf:"bytes,1,opt,name=raw"`
 }
 
+// OpenAPISchemaType is used by the kube-openapi generator when constructing
+// the OpenAPI spec of this type.
+//
+// See: https://github.com/kubernetes/kube-openapi/tree/master/pkg/generators
+func (_ JSON) OpenAPISchemaType() []string {
+	// TODO: return actual types when anyOf is supported
+	return []string{}
```

Comment:
what is the slice for? Multple types with quantors don't make much sense.

Reply:
> what is the slice for? 

go-openapi treats `type` as a slice because json schema treats type as a slice.

OpenAPI (v2 and v3), however, treats `type` as a string.

https://github.com/kubernetes/kubernetes/blob/27c327cd33db8c91f1c6ad6c701b0457c780bce2/vendor/github.com/go-openapi/spec/schema.go#L211

kube-openapi uses go-openapi and hence needs `[]string{}`

>  Multple types with quantors don't make much sense.

In our `JSONSchemaProps`, we treat it as a string:

https://github.com/kubernetes/kubernetes/blob/27c327cd33db8c91f1c6ad6c701b0457c780bce2/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types_jsonschema.go#L25